### PR TITLE
Chore/telemetry and shadow dom

### PR DESCRIPTION
**Pull request by vanbasten17**
## Description

* Upgrade to beta version
* Adapt `stylesLoaderConfig` to be used with shadow dom
* Add missing babel.config.js in typescript template
* Drop use of add-module-exports plugins (dropped in later versions of babel and were breaking webpack 5 in windows) https://babeljs.io/docs/en/v7-migration-api
* Add analytics script to every example
* Add missing workflows

### CODE REVIEW

File: blank-typescript/analytics.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+const os = require('os')
+
+function analytics_enabled() {
```
Comment (asastre) [inline blank-typescript/analytics.js]: Why not using camelCase?

File: blank-typescript/analytics.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    timestamp: new Date().toISOString(),
+  }
+}
+
+function track(event, properties = {}) {
+  if (
+    analytics_enabled() &&
+    analytics &&
+    credentials &&
+    credentials.analytics
```
Comment (asastre) [inline blank-typescript/analytics.js]: Where `analytics` and `credentials` come from?

**Review by asastre — CHANGES_REQUESTED** [review]
1 comment and 1 question before approval

File: blank-typescript/analytics.js (hunk 2 above)
Comment (vanbasten17) [reply blank-typescript/analytics.js]: thanks @asastre , i missed it! now fixed 👌

File: blank-typescript/analytics.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+const path = require('path')
+
+const BOTONIC_HOME_PATH = path.join(os.homedir(), '.botonic')
+const BOTONIC_CREDENTIALS_PATH = path.join(
+  BOTONIC_HOME_PATH,
+  'credentials.json'
+)
+
+let credentials
+const analytics = new Analytics('YD0jpJHNGW12uhLNbgB4wbdTRQ4Cy1Zu')
```
Comment (asastre) [inline blank-typescript/analytics.js:13]: Use constant

File: blank-typescript/analytics.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    timezone: Intl.DateTimeFormat().resolvedOptions().timeZone,
+    timestamp: new Date().toISOString(),
+  }
+}
+
+function initializeCredentials() {
+  if (!fs.existsSync(BOTONIC_HOME_PATH)) fs.mkdirSync(BOTONIC_HOME_PATH)
+  const anonymous_id = Math.round(Math.random() * 100000000)
+  fs.writeFileSync(
+    botonic_credentials_path,
```
Comment (asastre) [inline blank-typescript/analytics.js]: This should be `BOTONIC_CREDENTIALS_PATH`, right?

**Review by asastre — CHANGES_REQUESTED** [review]
1 suggestion and 1 thing I think it has to be changed

File: blank/analytics.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    platform: os.platform(),
+    arch: os.arch(),
+    timezone: Intl.DateTimeFormat().resolvedOptions().timeZone,
+    timestamp: new Date().toISOString(),
+  }
+}
+
+function initializeCredentials() {
+  if (!fs.existsSync(BOTONIC_HOME_PATH)) fs.mkdirSync(BOTONIC_HOME_PATH)
+  const anonymous_id = Math.round(Math.random() * 100000000)
```
Comment (dpinol) [inline blank/analytics.js:31]: why do we need the anonymous_id?

File: blank/analytics.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  }
+}
+
+const MODE = process.argv[2]
+try {
+  readCredentials()
+  if (MODE === 'serve') track('Served Botonic CLI')
+  if (MODE === 'train') track('Trained with Botonic train')
+} catch (e) {
+  console.warn('Credentials could not be loaded', e)
```
Comment (dpinol) [inline blank/analytics.js]: change the warn text here , as maybe the exception comes from the tracks

File: blank/analytics.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  }
+}
+
+function track(event, properties = {}) {
+  if (analyticsEnabled() && analytics && credentials && credentials.analytics) {
+    properties = {
+      ...properties,
+      ...getSystemInformation(),
+    }
+    analytics.track({
```
Comment (dpinol) [inline blank/analytics.js:57]: if this is async , return the promise and await when calling below so that the catch catches the exception

**Review by dpinol — CHANGES_REQUESTED** [review]
In the future, create a PR where you only modify 1 example.
WHen approved, open another PR where you copy/paste the changes to all examples.
It will make your life easier :-)

File: blank/analytics.js (hunk 7 above)
Comment (vanbasten17) [reply blank/analytics.js:57]: is not async

File: blank/analytics.js (hunk 6 above)
Comment (vanbasten17) [reply blank/analytics.js]: sure, changed! 👍

**Comment by vanbasten17** [comment]
Thanks for the advice @dpinol , will take into account! 👌

**Review by asastre — APPROVED** [review]
In a future PR maybe it would be nice to have all this logic into a class, what do you think?

**Comment by vanbasten17** [comment]
Yes @asastre , I think so, as this code is also available in cli, I would create a Botonic analytics package or similar and make use of them in both cli and templates! Then add it just as a dev dependency in the case of templates or as a dependency in the case of cli

**Review by dpinol — APPROVED** [review]

**Pull request merged**